api/src/services/checkouts/checkouts.js

```diff
@@ -43,7 +43,7 @@ export const createCheckoutSession = async ({ mode }) => {
     line_items: cartItems,
     payment_method_types: ['card'],
     mode: mode,
-    success_url: `http://localhost:8910/stripe-cart?success=true&sessionId={CHECKOUT_SESSION_ID}`,
-    cancel_url: `http://localhost:8910/stripe-cart?success=false`,
+    success_url: `${context.request.headers.referer}?success=true&sessionId={CHECKOUT_SESSION_ID}`,
+    cancel_url: `${context.request.headers.referer}?success=false`,
```

Comment:
The only alternative to this I could think of is using NODE_ENV to conditionally set the origin, but even then, that won't work for netlify deploy previews.